libmemscan/src/interactive.rs

```diff
@@ -0,0 +1,534 @@
+    pub fn from_bytes(bytes: &[u8], offset: usize, value_type: ValueType) -> Option<Self> {
+        if offset + value_type.size() > bytes.len() {
+            return None;
+        }
+        
+        let slice = &bytes[offset..offset + value_type.size()];
+        Some(match value_type {
+            ValueType::I8 => Value::I8(i8::from_le_bytes([slice[0]])),
+            ValueType::I16 => Value::I16(i16::from_le_bytes(slice.try_into().ok()?)),
+            ValueType::I32 => Value::I32(i32::from_le_bytes(slice.try_into().ok()?)),
+            ValueType::I64 => Value::I64(i64::from_le_bytes(slice.try_into().ok()?)),
+            ValueType::U8 => Value::U8(u8::from_le_bytes([slice[0]])),
+            ValueType::U16 => Value::U16(u16::from_le_bytes(slice.try_into().ok()?)),
+            ValueType::U32 => Value::U32(u32::from_le_bytes(slice.try_into().ok()?)),
+            ValueType::U64 => Value::U64(u64::from_le_bytes(slice.try_into().ok()?)),
+            ValueType::F32 => Value::F32(f32::from_le_bytes(slice.try_into().ok()?)),
+            ValueType::F64 => Value::F64(f64::from_le_bytes(slice.try_into().ok()?)),
+        })
+    }
+    
+    /// Convert value to bytes for writing to memory
+    pub fn to_bytes(&self) -> Vec<u8> {
+        match self {
+            Value::I8(v) => v.to_le_bytes().to_vec(),
+            Value::I16(v) => v.to_le_bytes().to_vec(),
+            Value::I32(v) => v.to_le_bytes().to_vec(),
+            Value::I64(v) => v.to_le_bytes().to_vec(),
+            Value::U8(v) => v.to_le_bytes().to_vec(),
+            Value::U16(v) => v.to_le_bytes().to_vec(),
+            Value::U32(v) => v.to_le_bytes().to_vec(),
+            Value::U64(v) => v.to_le_bytes().to_vec(),
+            Value::F32(v) => v.to_le_bytes().to_vec(),
+            Value::F64(v) => v.to_le_bytes().to_vec(),
+        }
+    }
+}
+
+/// Filter operation for comparing values
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum FilterOp {
+    /// Value equals a specific value
+    Equals,
+    /// Value is less than a specific value
+    LessThan,
+    /// Value is greater than a specific value
+    GreaterThan,
+    /// Value increased compared to previous scan
+    Increased,
+    /// Value decreased compared to previous scan
+    Decreased,
+    /// Value changed compared to previous scan
+    Changed,
+    /// Value unchanged compared to previous scan
+    Unchanged,
+}
+
+/// A memory address that matches the current filter criteria
+#[derive(Debug, Clone)]
+pub struct MatchedAddress {
+    /// Address in the target process
+    pub address: usize,
+    /// Current value at this address
+    pub current_value: Value,
+    /// Previous value (if available)
+    pub previous_value: Option<Value>,
+}
+
+/// Interactive memory scanner that maintains state between scans
+pub struct InteractiveScanner<'a> {
+    /// Target process handle
+    process: &'a ProcessHandle,
+    /// Memory regions being monitored
+    regions: Vec<MemoryRegion>,
```

Comment:
Use the MemoryDiff and snapshot capabilities instead of inventing new ones. If the existing is lacking, add the requirements that are missing!